cmd/terramate/cli/cli.go

```diff
@@ -935,40 +934,28 @@ func (c *cli) runOnStacks() {
 
 	logger.Info().Msg("Running on selected stacks")
 
-	failed := false
-
-	for _, stack := range orderedStacks {
-		cmd := exec.Command(c.parsedArgs.Run.Command[0], c.parsedArgs.Run.Command[1:]...)
-		cmd.Dir = stack.HostPath()
-		cmd.Env = os.Environ()
-		cmd.Stdin = c.stdin
-		cmd.Stdout = c.stdout
-		cmd.Stderr = c.stderr
-
-		logger := log.With().
-			Str("cmd", strings.Join(c.parsedArgs.Run.Command, " ")).
-			Stringer("stack", stack).
-			Logger()
+	err = run.Exec(
```

Comment:
Since run is more involved now with the signal handling I extracted it from the cli. It creates goroutines and changes signal handlers etc.